src/Combinatorics/Matroids/ChowRings.jl

```diff
@@ -0,0 +1,234 @@
+export chow_ring, augmented_chow_ring, select
+
+@doc Markdown.doc"""
+    chow_ring(M::Matroid, ring::MPolyRing=nothing; extended::Bool=false)
+
+Return the Chow ring of a matroid, optionally also with the simplicial generators and the polynomial ring.
+
+See [AHK18](@cite) and [BES19](@cite). 
+
+# Examples
+The following computes the Chow ring of the Fano matroid.
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = chow_ring(M);
+
+julia> R[1]*R[8]
+-x_{3,4,7}^2
+```
+
+The following computes the Chow ring of the Fano matroid including variables for the simplicial generators.
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = chow_ring(M, extended=true);
+
+julia> f = R[22] + R[8] - R[29]
+x_{1,2,3} + h_{1,2,3} - h_{1,2,3,4,5,6,7}
+
+julia> f==0
+true
+```
+
+The following computes the chow ring of the free matroid on three elements in a given graded polynomial ring.
+```jldoctest
+julia> M = uniform_matroid(3,3);
+
+julia> GR, _ = GradedPolynomialRing(QQ,["a","b","c","d","e","f"]);
+
+julia> R = chow_ring(M, GR);
+
+julia> hilbert_series_reduced(R)
+(t^2 + 4*t + 1, 1) 
+
+```
+"""
+function chow_ring(M::Matroid, ring::Union{MPolyRing,Nothing}=nothing; extended::Bool=false)
+    is_loopless(M) || error("Matroid has loops")
+    Flats = flats(M)
+    number_flats = length(Flats)
+    number_flats >= 2 || error("matroid has to few flats")
+    proper_flats = Flats[2:number_flats-1]
+
+    #construct polynomial ring and extract variables
+    if ring==nothing
+        # create variable names, indexed by the proper flats of M
+    replacements = ["["=>"{", "]"=>"}", ", "=>","]
+    fcts = [x->replace(x,y) for y in replacements]
+    var_names = [∘(fcts...)(string("x_",S)) for S in proper_flats]
+    #fix for the three lines above: usage of replace when 1.6 eol
+    #var_names = [replace(string("x_",S), "["=>"{", "]"=>"}", ", "=>",") for S in proper_flats] 
+        if extended
+            #add the variables for the simplicial generators
+        var_names = [var_names; [∘(fcts...)(string("h_",S)) for S in proper_flats]]
+        #fix same as above when 1.6 eol
+        #var_names = [var_names; [replace(string("h_",S), "["=>"{", "]"=>"}", ", "=>",") for S in [proper_flats;[Flats[number_flats]]]]] 
+        end
+            ring, vars = PolynomialRing(QQ, var_names, cached=false)
```

Comment:
```suggestion
        # create variable names, indexed by the proper flats of M
        var_names = ["x_{" * join(S, ",") * "}" for S in proper_flats]
        if extended
            #add the variables for the simplicial generators
            var_names = [var_names; ["h_{" * join(S, ",") * "}" for S in proper_flats]]
        end
        ring, vars = PolynomialRing(QQ, var_names, cached=false)
```
This should be more stable and makes it easier to read (and avoids some unicode symbols + many string replacements).

Reply:
Thank you!